vcd/datasource_vcd_resource_list.go

```diff
@@ -1158,9 +1165,15 @@ func genericResourceList(d *schema.ResourceData, resType string, ancestors []str
 			if ref.importId {
 				identifier = ref.id
 			}
-			list = append(list, fmt.Sprintf("terraform import %s.%s %s%s%s",
+			sanitizedName := ref.name
+			illegalHclNameCharsRegex := regexp.MustCompile(`[^a-zA-Z0-9_\-]+`)
+			if illegalHclNameCharsRegex.MatchString(ref.name) {
+				// Names can have special characters in VCD, but must not have in HCL resource names
+				sanitizedName = illegalHclNameCharsRegex.ReplaceAllString(ref.name, "_")
+			}
+			list = append(list, fmt.Sprintf("terraform import %s.%s '%s%s%s'",
```

Comment:
I'm a bit confused. Why do we need to quote the resource name if its name is already sanitised with underscores?

Reply:
In this case, it would not be the resource name the one in quotes, but the remote item in VCD to import (whose name cannot be changed).
For example, a catalog named _"a$b%c"_ would be `terraform import vcd_catalog.a_b_c 'a$b%c'`, where simple quotes prevent undesired side effects.